internal/controller/csiaddons/csiaddonsnode_controller.go

```diff
@@ -137,6 +138,40 @@ func (r *CSIAddonsNodeReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
+	// get the networkfenceclasses for the driver
+	nfclasses, err := r.getNetworkFenceClassesForDriver(ctx, &logger, csiAddonsNode)
+	if err != nil {
+		logger.Error(err, "Failed to get networkfenceclasses")
+	}
+
+	nfsc := make([]csiaddonsv1alpha1.NetworkFenceClientStatus, 0)
+	// get the clients to fence
+	for _, nfc := range nfclasses {
+		clients, err := getFenceClientDetails(ctx, newConn, &logger, nfc)
+		if err != nil {
+			logger.Error(err, "Failed to get clients to fence")
+			return ctrl.Result{}, err
+		}
+		logger.Info("Got clients to fence", "clients", clients)
+		if clients != nil {
+			nfcClientDetails := make([]csiaddonsv1alpha1.ClientDetail, 0)
+			for _, client := range clients.Clients {
+				logger.Info("Adding client to status", "client", client.Id, "cidrs", client.Cidrs)
+				nfcClientDetails = append(nfcClientDetails, csiaddonsv1alpha1.ClientDetail{
+					Id:    client.Id,
+					Cidrs: client.Cidrs,
+				})
+			}
+			nfsc = append(nfsc,
+				csiaddonsv1alpha1.NetworkFenceClientStatus{
+					NetworkFenceClassName: nfc.Name,
+					ClientDetails:         nfcClientDetails,
+				},
+			)
```

Comment:
Let's use a helper function here ? 

```suggestion
			nfsc = append(nfsc,
				extractNFClientStatus(&logger, nfc.Name, client.Clients),
			)
```

Reply:
IMO there is no gain in doing it because its not used in multiple places nor its a complex check, its just a for loop

Reply:
This part has a loop and itself inside a if condition, which in turn is inside a loop.
It looks messy.

Reply:
I agree with Rakshith. The function is already very large, and with this change it spans > 120 lines. Please move it out into a helper to make it a little more modular and easier to understand/maintain in the future.

Reply:
`getNetworkFenceClientStatus()` or something would be nice.